src/cs/ubc/ca/graphTeX/ast/Node.java

```diff
@@ -3,12 +3,22 @@
 import cs.ubc.ca.graphTeX.visitor.Visitor;
 
 public class Node extends AST {
-    String displayValue;
-    String refId;
-    String optLabel;
+    public String displayValue;
+    public String refId;
+    public String optLabel;
+
+    public Node(String displayValue, String refId) {
+        this.displayValue = displayValue;
+        this.refId = refId;
+    }
 
     @Override
     public <R> R accept(Visitor<R> v) {
         return v.visit(this);
     }
+
+    @Override
+    public String toString() {
```

Comment:
What's this used for? Also are we making id required now for a Node?

Reply:
No. If you take a look at the parser, any node declaration preceded by STAR creates a Node with the id and value set to be the same (so implicitly setting the id... or value depending on how you look at it).